### PR TITLE
[papi,dashboard] split `ListTeams` into `ListTeams`, `GetTeamInvitation` and `ListTeamMembers`

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

Previous Public API `ListTeams` will list teams of user and list all members + generate invite link for all those teams, it's very slow and not that necessary. This PR is about to split `ListTeams` API into APIs and call them when necessary.

- Split papi `ListTeams`(member, invite id included) into
    - `ListTeams` (organization info id, name, slug only)
    - `ListTeamMembers`
    - `GetTeamInvitation`
- Change where we use this API to prevent build failed
- Change dashboard usage

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at a5eb7a3</samp>

This pull request refactors and enhances the dashboard and the public API components to use new hooks and data sources for fetching and displaying information about the current organization and its members. It also removes some unused code and fields related to the teams feature. The main files affected are `components/dashboard/src/data/organizations/org-members-info-query.ts`, which defines the `useOrgMembersInfoQuery` hook, and `components/public-api-server/pkg/apiv1/team.go`, which adds new methods to the `TeamService` type.

</details>

## Related Issue(s)

Related EXP-775

## How to test

- Go to preview env, organization related stuff should work like before (should try different role `owner`, `member`, join my org with link https://hw-papi-list-team.preview.gitpod-dev.com/orgs/join?inviteId=14721f4a-143e-43d1-abf1-5df1a584374b )
    - [x] Members invite
    - [x] Org settings
    - [x] Member role changing
    - [x] Member list
- `ListTeams` API should not respond with `members` and `invitation` anymore

**✅ ListTeamMembers should not block homepage (/workspace) load**
- Open Gitpod with this PR
- Go to `components/dashboard`
- Add sleep 20 seconds to `useOrgMembersInfoQuery` (search useOrgMembersInfoQuery)
- Run `yarn telepresence` to inject dashboard directly to preview env
- Refresh preview env homepage  

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - hw-papi-list-team</li>
	<li><b>🔗 URL</b> - <a href="https://hw-papi-list-team.preview.gitpod-dev.com/workspaces" target="_blank">hw-papi-list-team.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - hw-papi-list-team-gha.18414</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-hw-papi-list-team%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-core-dev" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Comment by easyCZ** [comment]
Worth noting, that you have no guarantee on the rollout ordering. This means that the dashboard can be rolled out first, but at that point, it has no corresponding APIs implemented. Ideally, changes like this would first be introduced on the API, validated, and only then introduced into the clients.

File: components/public-api/gitpod/experimental/v1/teams.proto [hunk 1]
```diff
@@ -142,6 +157,16 @@ message ResetTeamInvitationResponse {
...
 }
 
+message ListTeamMembersRequest {
+    // team_id is the ID of the team that contains the members to list
+    string team_id = 1;
+}
+
+message ListTeamMembersResponse {
+    // members are the team members of this Team
+    repeated TeamMember members = 1;
```
Comment (svenefftinge) [inline components/public-api/gitpod/experimental/v1/teams.proto:167]: 👍 let's tackle [this](https://linear.app/gitpod/issue/EXP-778/members-list-should-be-paginated) in a follow up.

Comment (mustard-mh) [reply components/public-api/gitpod/experimental/v1/teams.proto:167]: Not sure if pagination is worth doing for API, that our search on dasbboard will search for user name and email both, then we need to do `debounce` on dashboard and sql `<fields> like %?%` on server

**Comment by mustard-mh** [comment]
> Worth noting, that you have no guarantee on the rollout ordering. This means that the dashboard can be rolled out first, but at that point, it has no corresponding APIs implemented. Ideally, changes like this would first be introduced on the API, validated, and only then introduced into the clients.

@easyCZ Thank you for point out, then I'll split into PRs to make sure not compatibility issues

File: components/public-api/gitpod/experimental/v1/teams.proto (hunk 1 above)
Comment (svenefftinge) [reply components/public-api/gitpod/experimental/v1/teams.proto:167]: In dedicated we can easily have hundreds of organization members

**Comment by easyCZ** [comment]
Also worth considering to add pagination to any List calls. The risk with not doing so is that a client expects to get the full list, and once you introduce it, even in a backwards compatbile way, the client will never actually go and fetch the next page. Perhaps less risk here as we control the dashboard, but we should try to get into the habit of this.

**Comment by mustard-mh** [comment]
Close since https://github.com/gitpod-io/gitpod/pull/18908#issuecomment-1759050859

**Pull request closed without merging**